File: ruby/scrape.rb
require 'rubygems' 
require 'tweetstream'

TweetStream.configure do |config|
  config.consumer_key       = ''
  config.consumer_secret    = ''
  config.oauth_token        = ''
  config.oauth_token_secret = ''
  config.auth_method        = :oauth
end

start = Time.now

TweetStream::Client.new.sample do |status|
  puts JSON.generate(status.attrs)

  # stop after 1 hour
  current = Time.now
  if current - start >= 60 * 60 then
     break
  end
end

File: python/scrape.py
import sys, os, urllib2, tweetstream, json
from datetime import datetime as dt

CONSUMER_KEY = ""
CONSUMER_SECRET = ""
ACCESS_TOKEN_KEY =  ""
ACCESS_TOKEN_SECRET =  ""

stream=tweetstream.SampleStream

start=dt.now()

with stream(CONSUMER_KEY, CONSUMER_SECRET, ACCESS_TOKEN_KEY, ACCESS_TOKEN_SECRET) as stream:
	while 1==1:
	        try: 
   			for tweet in stream:
				print json.dumps(tweet)
				now = dt.now()
				if (now - start).seconds > 3600:
					break
		except:
			now = dt.now()
                        if (now - start).seconds > 3600:
                        	break
			else:
				continue
